### ISSUE TITLE
Add ability to visualize spots to fit grains results dialog

### ISSUE THREAD

**Issue by psavery** [HEXRD/hexrdgui#945]
In the fit grains results viewer, add a right-click context menu for the rows of the table to view the spots for the selected grain. This would probably pop up a new dialog with a canvas in it with the visualization.

@joelvbernier It might be useful to get some example code for the visualization.

We will also need to figure out if we are currently capturing the data we need to visualize this, or if we need to capture more variables. `pull_spots` is called via multiprocessing as a part of fit grains [here](https://github.com/HEXRD/hexrd/blob/e1f9b320d24b807ef7d6e1da70217244a6bf72c0/hexrd/fitgrains.py#L100). I'm wondering if [the results from fit grains](https://github.com/HEXRD/hexrd/blob/e1f9b320d24b807ef7d6e1da70217244a6bf72c0/hexrd/fitgrains.py#L402) includes the spots data that we will use for visualization.

If not, we will either need to add the spots data to the results from `fit_grains`, or if it is easier, we can:

1. Require that the user write out the spots files (see #944) at the time of grain fitting. We can save a variable that indicates whether or not they did so.
2. Then, when the user right-clicks on a grain, we can check the `analysis_dir` to see if there are spots files, and if there are, load the spots file corresponding to the selected grain and plot the data.

This would mean that the menu path "File"->"Open"->"Grain Fitting Results" will **not** allow the user to view spots data, because they did not run through the fit grains workflow (where the spots data would be generated). That is, unless we standardize things somehow, like requires a `spots` directory to be next to the `grains.out` file in the directory structure... or make an h5 file that includes the `grains.out` data and spots data...

**Comment by bnmajor** [comment]
@joelvbernier Do you have any example code for the visualization or an outline of what exactly we should be visualizing? Also it looks like the [results from fit grains](https://github.com/HEXRD/hexrd/blob/e1f9b320d24b807ef7d6e1da70217244a6bf72c0/hexrd/fitgrains.py#L402) does **not** return the spots data. Would it be preferred that `pull_spots` is updated to support returning the spots as well as writing them out, or would it be sufficient to require that the user first save the spots in order to visualize them as @psavery suggested?

**Comment by joelvbernier** [comment]
I sent in Slack, I believe @psavery had got it running.

**Comment by psavery** [comment]
Yes, I do have one working. However, we're going to need to make a way for `fit_grains()` to return the spots.

Looks like using the [`save_spots_list` parameter](https://github.com/HEXRD/hexrd/blob/2b6f1445b25db30ab6007a111d0aa1bdbd023742/hexrd/instrument.py#L1403) on `pull_spots()` may be a start.
